File: task-03/Go/Sub-Task 3/subtask3.go
package main
import (
    "fmt"
    "strings"
)
func main() {
    var n int
    fmt.Print("Enter a number: ")
    fmt.Scan(&n)
    for i := 0; i < n; i++ {
        fmt.Println(strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1))
    }
    for i := n-2; i >= 0; i-- {
        fmt.Println(strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1))
    }
}

File: task-03/Rust/Sub-Task 3/subtask3.rs
use std::io;

fn print_diamond(n: usize) {
    for i in 0..n {
        println!("{:width$}{}", "", "*".repeat(2 * i + 1), width = n - i - 1);
    }
    for i in (0..n-1).rev() {
        println!("{:width$}{}", "", "*".repeat(2 * i + 1), width = n - i - 1);
    }
}

fn main() {
    let mut input = String::new();
    println!("Enter a number: ");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Please type a number!");
    print_diamond(n);
}